src/main/java/carRacing/domain/Car.java

```diff
@@ -0,0 +1,37 @@
+package carRacing.domain;
+
+import java.util.Random;
+
+public class Car {
+    private int score;
+
+    public Car(){
+        this.score = 0;
+    }
+
+    public Car(Car car){
+        this.score = car.score;
+    }
+
+    public Car raceOrStay(Car car){
+        Car carAfterAction = new Car(car);
+        Random random = new Random();
+        if(random.nextInt(10)>=4) {
+            carAfterAction.score+=1;
+        }
+        return carAfterAction;
```

Comment:
```suggestion
         if(new Random().nextInt(10)>=4) {
             return new Car(score + 1);
         }
         return this;
```
위에 제안한 주생성자 호출방식으로 적용한다면 이렇게 바꿔볼 수 있을것 같습니다.
추가로 `Random` 을 매번 생성하면 편향된 결과가 나올 수 있습니다.
https://rules.sonarsource.com/java/RSPEC-2119

Random 객체를 static 으로 선언하거나 좀 더 고성능이 필요한경우 [ThreadLocalRandom](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html)  을 사용합니다.

Reply:
좋은 사이트 추천 정말 감사드립니다. 인텔리제이에서 소나린트를 사용하고 있으면서도 저런 사이트를 찾아볼 생각은 하지 못했네요ㅠㅠ